docbook/reference/en/en-US/modules/Reactive.xml

```diff
@@ -1,63 +1,810 @@
+          <ulink url="https://community.oracle.com/docs/DOC-1006738">https://community.oracle.com/docs/DOC-1006738/</ulink>:
+          (Java JDK 9+)
+          </listitem>
+    </orderedlist>
+    
+    <para>
+       RESTEasy currently supports RxJava (deprecated) and RxJava2.
+    </para>
+    
+</sect1>
+
+<sect1>
+<title>Pluggable reactive types: RxJava 2 in RESTEasy</title>
+
+    <para>
+       JAX-RS 2.1 doesn't require support for any Reactive Streams implementations, but it does allow
+       for extensibility to support various reactive libraries.
+       RESTEasy's optional modules <code>resteasy-rxjava1</code> and <code>resteasy-rxjava2</code> 
+       add support for <link linkend="https://github.com/ReactiveX/RxJava">RxJava 1 and 2</link>.
+       [Only <code>resteasy-rxjava2</code> will be discussed here, since <code>resteasy-rxjava1</code> is
+       deprecated, but the treatment of the two are quite similar.]
+    </para>
+    
+    <para>
+       In particular, <code>resteasy-rxjava2</code>
+       contributes support for reactive types <classname>io.reactivex.Single</classname>,
+       <classname>io.reactivex.Flowable</classname>, and <classname>io.reactivex.Observable</classname>.
+       Of these, <classname>Single</classname> is similar to <classname>CompletionStage</classname> in that
+       it holds at most one potential value. <classname>Flowable</classname> implements 
+       <classname>io.reactivex.Publisher</classname>, and <classname>Observable</classname> is very
+       similar to <classname>Flowable</classname> except that it doesn't support backpressure.
+       So, if you import <code>resteasy-rxjava2</code>, you can just start returning these reactive types from your
+       resource methods on the server side and receiving them on the client side.
+    </para>
+    
+<sect1>
+<title>Server side</title>
+    <para>
+       For example, give the class <classname>Thing</classname>, which can be represented in JSON:
+    </para>
+    
+<programlisting>
+public class Thing {
+
+   private String name;
+
+   public Thing() {
+   }
+
+   public Thing(String name) {
+      this.name = name;
+   }
+   ...
+}
+</programlisting>
+
+    <para>the method <methodname>postThingList()</methodname> in the following is a valid resource method:
+    </para>
+...
+<programlisting>
+@POST
+@Path("post/thing/list")
+@Produces(MediaType.APPLICATION_JSON)
+@Stream
+public Flowable&lt;List&lt;Thing&gt;&gt; postThingList(String s) {
+   return buildFlowableThingList(s, 2, 3);
+}
+
+static Flowable&lt;List&lt;Thing&gt;&gt; buildFlowableThingList(String s, int listSize, int elementSize) {
+   return Flowable.create(
+      new FlowableOnSubscribe&lt;List&lt;Thing&gt;&gt;() {
+
+         @Override
+         public void subscribe(FlowableEmitter&lt;List&lt;Thing&gt;&gt; emitter) throws Exception {
+            for (int i = 0; i &lt; listSize; i++) {
+               List&lt;Thing&gt; list = new ArrayList&lt;Thing&gt;()&lt;

```

Comment:
`;`char should be at the end of this line instead of `&lt;` ...

Reply:
Aaarrgh. Yes, thanks.